NewsFeed/views/CCBRNewsViewController.m

```diff
@@ -83,9 +86,13 @@ - (void)viewDidLoad {
 }
 
 - (void)updateUI {
-    self.collectionView.hidden = self.viewModel.collectionViewHidden;
     self.errorMessageLabel.hidden = self.viewModel.errorMessageLabelHidden;
-    [self.collectionView reloadData];
+    self.errorMessageLabel.text = self.viewModel.errorMessageDescription;
+    
+    [self.collectionContainerView bringSubviewToFront:self.errorMessageLabel];
+    
+    self.collectionView.hidden = self.viewModel.collectionViewHidden ;
+    if(self.viewModel.shouldReloadGridData) [self.collectionView reloadData];
```

Comment:
This code is not optimized because all collection view cells will be reloaded instead of being inserted.